contracts/mocks/EIP712External.sol

```diff
@@ -22,7 +22,7 @@ contract EIP712External is EIP712 {
         require(recoveredSigner == signer);
     }
 
-    function getChainId() external pure returns (uint256 chainId) {
+    function getChainId() external view returns (uint256 chainId) {
         // solhint-disable-next-line no-inline-assembly
         assembly {
             chainId := chainid()
```

Comment:
0.8.0 provides this as a language feature: `block.chainid`

Reply:
Solved on 768d92ca9c5755af864d90a51ffa2531e334d36d.